pulsar-rpc-contrib/src/main/java/org/apache/pulsar/rpc/contrib/server/RequestListener.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.pulsar.rpc.contrib.server;
+
+import static org.apache.pulsar.rpc.contrib.common.Constants.REPLY_TOPIC;
+import static org.apache.pulsar.rpc.contrib.common.Constants.REQUEST_TIMEOUT_MILLIS;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.client.api.MessageListener;
+
+/**
+ * Handles incoming Pulsar messages, processes them using a specified function, and sends replies using a
+ * {@link ReplySender}. This listener is typically used on the server side of a Pulsar RPC implementation.
+ *
+ * @param <T> the type of the request messages
+ * @param <V> the type of the response messages
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class RequestListener<T, V> implements MessageListener<T> {
```

Comment:
If you do not need to expose this class to the business, you can use package-level private.

Reply:
Done.